src/request.ts

```diff
@@ -28,12 +31,55 @@ export function setRequestLogger(logger: Logger): void {
 	requestLogger = logger;
 }
 
-async function _request({
-	endpoint,
-	requestBody,
-	headers: requestHeaders,
-	...options
-}: RequestOptions): Promise<unknown> {
+/**
+ * Internal function that handles retry logic for NUT-19 cached endpoints.
+ * Non-cached endpoints are executed directly without retries.
+ */
+async function requestWithRetry(options: RequestOptions): Promise<unknown> {
+	const { ttl, cached_endpoints, endpoint } = options;
```

Comment:
Memo: references to `cached_endpoint` option in this function would need updating to `cachedEndpoint` if we adopt the camelCase suggestion in Nut19Policy.